### PR TITLE
VisIt v2.10.1

### PR DESCRIPTION
**Pull request by dhandeo**
- Adds VisIt v2.10.1
- Useful as is, but can be enhanced later by adding more variants and more dependencies from spack

### CODE REVIEW

**Comment by tgamblin** [comment]
@cyrush :tada:

**Comment by cyrush** [comment]
looks like a good start -- I think Silo support should be there as a bare minimum. The Silo spack package seems ready to use.

File: var/spack/repos/builtin/packages/visit/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+class Visit(Package):
+    """VisIt is an Open Source, interactive, scalable, visualization, animation and analysis tool."""
+    homepage = "https://wci.llnl.gov/simulation/computer-codes/visit/"
+    url = "http://portal.nersc.gov/project/visit/releases/2.10.1/visit2.10.1.tar.gz"
+
+    version('2.10.1', '3cbca162fdb0249f17c4456605c4211e')
+
+    depends_on("vtk@6.1.0~opengl2")
+    depends_on("qt@4.8.6")
```
Comment (mathstuf) [inline var/spack/repos/builtin/packages/visit/package.py]: Can this just be `@4` or maybe `@4.8.6:4.9`.

**Comment by tgamblin** [comment]
@dhandeo Is this ready to merge as-is or do you have some more commits?

**Comment by dhandeo** [comment]
@tgamblin, I am working on building visit with silo as requested, and that is still underway (Builds fails at 89% during target ECaleDatabase_ser). That could be a separate pr when ready if you merge this.

**Comment by dhandeo** [comment]
@tgamblin @cyrush builds VisIt with silo. I can add @4.8.6:4.9 for qt versions as @mathstuf  suggested, however VisIt seems to be very strict in terms of versions of required dependencies and I tested only with 4.8.6. Please let me now.

**Comment by cyrush** [comment]
@dhandeo Looks good! Thanks for doing this. 

We are happy to use use very specific versions, that is the way we build things in build_visit.

Some of that is due to build complexity on various platforms (something that spack and help with) and some of it is simply due to worry of new bugs, interface changes, with untested versions.

**Comment by adamjstewart** [comment]
Version 2.10.2 was recently released. Can you add that to this?

File: var/spack/repos/builtin/packages/visit/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    depends_on("python")
+    depends_on("hdf5")  # silo seems to need it
```
Comment (davydden) [inline var/spack/repos/builtin/packages/visit/package.py]: if `silo` needs it it should be a dependency in `silo` only?

File: var/spack/repos/builtin/packages/visit/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    depends_on("silo+shared")
+
+    def install(self, spec, prefix):
+        with working_dir('spack-build', create=True):
+
+            feature_args = std_cmake_args[:]
+            feature_args = ["-DVTK_MAJOR_VERSION=6",
+                            "-DVTK_MINOR_VERSION=1",
+                            "-DCMAKE_INSTALL_PREFIX:PATH=%s" % spec.prefix,
```
Comment (davydden) [inline var/spack/repos/builtin/packages/visit/package.py]: this is most likely there in `std_cmake_args`, you can check `spack_build.out` to be sure.

File: var/spack/repos/builtin/packages/silo/package.py [hunk 4]
```diff
@@ -21,6 +23,8 @@ def install(self, spec, prefix):
         config_args = [
             '--enable-fortran' if '+fortran' in spec else '--disable-fortran',
             '--enable-silex' if '+silex' in spec else '--disable-silex',
+            '--enable-shared' if '+shared' in spec else '--disable-shared',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/silo/package.py]: can this not build both shared and static at the same time?

File: var/spack/repos/builtin/packages/visit/package.py (hunk 3 above)
Comment (cyrush) [inline var/spack/repos/builtin/packages/visit/package.py]: just to clarify, you are talking about the CMAKE_INSTALL_PREFIX line?

Comment (davydden) [inline var/spack/repos/builtin/packages/visit/package.py]: yes

**Comment by dhandeo** [comment]
@davydden @cyrush @tgamblin 
Added v2.10.2 and correctly extending std_cmake_args, that resolves all other concerns on this pr.
 
If these needs rebase on top of latest develop, I will do that tomorrow

**Comment by tgamblin** [comment]
@dhandeo: thanks!  initial visit merged.

**Comment by cyrush** [comment]
@tgamblin @dhandeo thanks for this!

**Pull request merged**